src/eventcontroller/EventController.ts

```diff
@@ -1,16 +1,34 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import EventObserver from '../eventobserver/EventObserver';
+import EventReporter from '../eventreporter/EventReporter';
 import AudioVideoEventAttributes from './AudioVideoEventAttributes';
 import DeviceEventAttributes from './DeviceEventAttributes';
 import EventName from './EventName';
 import MeetingHistoryState from './MeetingHistoryState';
 
 /**
- * [[EventController]] keeps track of events and notifies audio-video observers.
- * An event describes the success and failure conditions for the meeting session.
+ * [[EventController]] keeps track of a list of event observers and notifies them of SDK events.
+ * An event can describe success or failure of SDK components that make use of this controller.
+ * Example: The success and failure for a meeting session.
  */
 export default interface EventController {
+  /**
+   * Adds an observer for event published to this controller.
+   */
+  addObserver(observer: EventObserver): void;
+
+  /**
+   * Removes an observer for event published to this controller.
+   */
+  removeObserver(observer: EventObserver): void;
+
+  /**
+   * Passes each observer of this controller to a given function.
+   */
+  forEachObserver(observerFunc: (observer: EventObserver) => void): void;
```

Comment:
Does this need to be public?

Reply:
Perhaps not, but because the same function on `AudioVideoController` is public builder might have been calling their own observerFunction that uses `eventDidRecieve` so it's best to make it public to avoid difficult migration where they would have to then make their own `EventController`

Reply:
Yeah we can try just changing it to private for now, I'll say the functionality has been deprecated for eventDidRecieve but if a builder wants the functionality they can request it to be made public again.